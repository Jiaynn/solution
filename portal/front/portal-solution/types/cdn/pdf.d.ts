declare let pdfExport: string

declare module '*.pdf' {
  export = pdfExport
}
