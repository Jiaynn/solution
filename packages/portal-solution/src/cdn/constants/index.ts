// 启用动态加速工单地址
export const enableDynamicURL = 'https://support.qiniu.com/tickets/new/form?category=%E5%85%B6%E4%BB%96%E7%B1%BB%E5%92%A8%E8%AF%A2&space=%E8%9E%8D%E5%90%88%20CDN'

// 证书配置提交工单 URL
export const certConfigHelpURL = 'https://support.qiniu.com/tickets/new/form?category=%E8%AF%81%E4%B9%A6%E9%97%AE%E9%A2%98&space=CDN'
