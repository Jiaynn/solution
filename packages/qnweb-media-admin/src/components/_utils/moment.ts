import moment from 'moment';

/**
 * 格式化duration
 * @param duration 时长，单位毫秒(ms)
 */
export const formatDuration = (duration?: number) => {
  return moment.utc(duration).format('HH:mm:ss');
};

/**
 * 格式化时间
 * @param datetime
 * @param format
 */
export const formatDatetime = (datetime: moment.MomentInput, format?: string) => {
  return moment(datetime).format(format || 'YYYY-MM-DD HH:mm:ss');
};

/**
 * 毫秒=>秒
 * @param value
 */
export const millisecondToSecond = (value: number) => {
  return value / 1000;
};

/**
 * 秒=>毫秒
 * @param value
 */
export const secondToMillisecond = (value: number) => {
  return value * 1000;
};
