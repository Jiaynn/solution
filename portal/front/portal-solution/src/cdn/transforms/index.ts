import { QueryParams } from 'qn-fe-core/utils'

export function atoi(numberText: string, valueForInvalidInput = 0): number {
  const parsed = parseInt(numberText, 10)
  return Number.isNaN(parsed) ? valueForInvalidInput : parsed
}

export function atof(numberText: string, valueForInvalidInput = 0): number {
  const parsed = parseFloat(numberText)
  return Number.isNaN(parsed) ? valueForInvalidInput : parsed
}

export function getIntValue(e: React.FormEvent<any>, valueForInvalidInput = 0) {
  return atoi((e.target as any).value, valueForInvalidInput)
}

export function getFloatValue(e: React.FormEvent<any>, valueForInvalidInput = 0) {
  return atof((e.target as any).value, valueForInvalidInput)
}

export function trimAndFilter(lines: string[]): string[] {
  return (lines || []).map(
    line => line.trim()
  ).filter(
    line => !!line
  )
}

export function splitLines(linesStr: string | null | undefined): string[] {
  if (!linesStr) {
    return []
  }
  return linesStr.split('\n')
}

export function filterQuery(query?: QueryParams) {
  if (!query) {
    return
  }
  return Object.keys(query).reduce(
    (filtered, key) => (
      (query || {})[key]
      ? { ...filtered, [key]: (query || {})[key] }
      : filtered
    ), {} as QueryParams
  )
}

export function humanizeOnoff(isOn: boolean) {
  return isOn ? '开启' : '关闭'
}
