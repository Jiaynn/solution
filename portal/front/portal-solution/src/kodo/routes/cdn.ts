/**
 * @file CDN route TODO: 站内外链与内链分开
 * @author yinxulai <me@yinxulai.com>
 */

// CDN 概览
export function getCDNOverviewPath() {
  return '/cdn/overview'
}

// CDN 业务的域名创建
export function getCDNCreateDomainPath(bucketName: string) {
  const query = bucketName ? `?bucket=${encodeURIComponent(bucketName)}&fixBucket` : ''
  return `/cdn/domain/create${query}`
}

// CDN 业务的域名列表
export function getCDNDomainListPath() {
  return '/cdn/domain'
}

export function getCDNDomainDetailPath(domain: string) {
  return `/cdn/domain/${domain}`
}

export function getCDNCreateBucketDomainPath(bucketName: string) {
  return `/cdn/domain/create?bucket=${bucketName}&fixBucket`
}

export function getCDNRefreshPath() {
  return '/cdn/refresh-prefetch'
}

export function getCDNRefreshLogsPath() {
  return '/cdn/refresh-prefetch?tab=showLogs'
}
