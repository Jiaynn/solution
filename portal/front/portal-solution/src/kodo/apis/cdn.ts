/**
 * @file config api
 * @description region 相关的接口
 * @author lizhifeng <lizhifeng@qiniu.com>
 */

import autobind from 'autobind-decorator'
import { injectable } from 'qn-fe-core/di'
import { ApiException } from 'qn-fe-core/client'

import { ProxyClientV2 } from 'portal-base/common/apis/proxy'

import { proxy } from 'kodo/constants/apis'

interface CdnBaseResponse {
  code: number
  error: string
  [k: string]: any
}

export interface RefreshCdnSurplus extends CdnBaseResponse {
  urlQuotaDay: number // url 刷新当日限额
  urlSurplusDay: number // url 刷新当日余量
  dirQuotaDay: number // 目录刷新当日限额
  dirSurplusDay: number // 目录刷新当日余量
  quotaDay: number // 预取当日限额
  surplusDay: number // 预取当日剩余量
}

interface RefreshCdnOptions {
  product: 'cdn'
  urls: string[]
}

export class CdnApiException extends ApiException {
  constructor(
    public code: number,
    public payload: CdnBaseResponse,
    message?: string,
    cause?: unknown
  ) {
    super(
      'RefreshPrefetchApiException',
      code,
      message,
      undefined,
      cause
    )
  }
}

const messageMap = new Map<number, string>([
  [400031, '无效的 URL'],
  [400034, '刷新域名个数超过限制'],
  [400037, 'URL 已在队列中，无需重复提交'],
  [400039, '请求次数过多']
])

function produceResult<T extends CdnBaseResponse>(payload: T) {
  if (payload && typeof payload === 'object' && 'code' in payload && payload.code !== 200) {
    const message = messageMap.get(payload.code) || payload.error
    throw new CdnApiException(payload.code, payload, message)
  }

  return payload
}

@autobind
@injectable()
export class CdnApis {
  constructor(
    private proxyClient: ProxyClientV2
  ) {}

  getRefreshCdnSurplus(): Promise<RefreshCdnSurplus> {
    return this.proxyClient.get<RefreshCdnSurplus>(proxy.refreshCdnSurplus, {})
      .then(produceResult)
  }

  async refreshCdn(params: RefreshCdnOptions): Promise<void> {
    await this.proxyClient.post<CdnBaseResponse>(proxy.refreshCdn, params)
      .then(produceResult)
  }
}
