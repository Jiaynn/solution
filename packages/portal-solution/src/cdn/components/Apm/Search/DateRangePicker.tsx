import React from 'react'
import { observer } from 'mobx-react'
import moment from 'moment'

import DatePicker from 'react-icecream/lib/date-picker'

import { getLatestDuration } from 'cdn/transforms/datetime'

import { APM_START_DATE } from 'cdn/constants/apm'

type Moment = moment.Moment
const RangePicker = DatePicker.RangePicker

export interface IDateRangePickerProps {
  value: [Moment, Moment],
  onChange: (dates: [Moment, Moment]) => void
}

export default observer(function DateRangePicker(props: IDateRangePickerProps) {
  const now = moment()
  const shortcuts: Record<string, [moment.Moment, moment.Moment]> = {
    今天: getLatestDuration(now, 0, 'day'),
    昨天: [moment(now).startOf('day').subtract(1, 'day'), moment(now).startOf('day').subtract(1, 'second')],
    最近7天: getLatestDuration(now, 7, 'day'),
    本月: [moment(now).startOf('month'), now],
    最近30天: getLatestDuration(now, 30, 'day')
  }
  const ranges = Object.keys(shortcuts).filter(key => {
    const duration = shortcuts[key]
    return !duration[0].isBefore(APM_START_DATE)
  }).reduce((rangeMap, key) => {
    rangeMap[key] = shortcuts[key]
    return rangeMap
  }, {} as Record<string, [moment.Moment, moment.Moment]>)
  const disabledDate = (current?: moment.Moment) => (
    current == null
    ? false
    : (
      current.clone().startOf('day').isAfter(now)
        || current.clone().isBefore(APM_START_DATE)
    )
  )

  return (
    <RangePicker
      format="YYYY-MM-DD HH:mm"
      showTime={{ format: 'HH:mm' }}
      allowClear={false}
      value={props.value as any}
      onChange={dates => props.onChange(dates as any)}
      onOk={dates => props.onChange(dates as any)}
      ranges={ranges}
      disabledDate={disabledDate}
    />
  )
})
