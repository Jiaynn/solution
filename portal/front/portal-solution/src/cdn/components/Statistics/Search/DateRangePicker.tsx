import React from 'react'
import { observer } from 'mobx-react'
import moment from 'moment'

import { useTranslation } from 'portal-base/common/i18n'
import DatePicker, { PickerProps } from 'react-icecream/lib/date-picker'

import * as messages from './messages'
import { getLatestDuration } from 'cdn/transforms/datetime'

type Moment = moment.Moment
const RangePicker = DatePicker.RangePicker

export interface IDateRangePickerProps {
  value: [Moment, Moment]

  onChange: (dates: [Moment, Moment]) => void

  /** 限制起始时间从当前时间往前推 n 天 */
  disableDays?: number
}

export default observer(function DateRangePicker(props: IDateRangePickerProps) {
  const t = useTranslation()
  const now = moment()

  const shortcuts = {
    [t(messages.today)]: getLatestDuration(now, 0, 'day'),
    [t(messages.yesterday)]: getLatestDuration(now, 1, 'day'),
    [t(messages.lastNDays, 7)]: getLatestDuration(now, 7, 'day'),
    [t(messages.thisMonth)]: [moment(now).startOf('month'), now],
    [t(messages.lastNDays, 30)]: getLatestDuration(now, 30, 'day')
  }

  const disabledDate: PickerProps['disabledDate'] = current => (current == null
    ? false
    : (
        current.unix() > now.unix()
        || (props.disableDays != null && current.unix() < now.clone().subtract(props.disableDays, 'day').unix())
      )
  )

  return (
    <RangePicker
      allowClear={false}
      value={props.value}
      onChange={props.onChange}
      ranges={shortcuts as any}
      style={{ width: 240 }}
      disabledDate={disabledDate}
    />
  )
})
