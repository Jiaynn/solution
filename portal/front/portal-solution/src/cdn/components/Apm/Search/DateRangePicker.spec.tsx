import React from "react";
import moment from "moment";

import { RendererUtils as Renderer } from "cdn/test";
import DateRangePicker from "./DateRangePicker";

const noop = () => null;

it("renders correctly", () => {
  const renderer = new Renderer();
  const tree = renderer
    .createWithAct(
      <DateRangePicker
        value={[moment("2018-08-09"), moment("2018-08-10")]}
        onChange={noop}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
