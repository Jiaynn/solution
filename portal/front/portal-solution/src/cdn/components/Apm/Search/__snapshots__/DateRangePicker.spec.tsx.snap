// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<span
  className="ant-calendar-picker"
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "width": 305,
    }
  }
  tabIndex={0}
>
  <span
    className="ant-calendar-picker-input ant-input"
    onClick={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onTouchStart={[Function]}
  >
    <input
      className="ant-calendar-range-picker-input"
      placeholder="Start date"
      readOnly={true}
      tabIndex={-1}
      value="2018-08-09 00:00"
    />
    <span
      className="ant-calendar-range-picker-separator"
    >
       
      ~
       
    </span>
    <input
      className="ant-calendar-range-picker-input"
      placeholder="End date"
      readOnly={true}
      tabIndex={-1}
      value="2018-08-10 00:00"
    />
    <i
      aria-label="icon: calendar"
      className="anticon anticon-calendar ant-calendar-picker-icon"
    >
      <svg
        aria-hidden="true"
        className=""
        data-icon="calendar"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
        />
      </svg>
    </i>
  </span>
</span>
`;
